### PR TITLE
Add support for #rgba, and #rrggbbaa rgba color parser

### PR DESCRIPTION
**Pull request by spsDrop**
Closes [2296](https://github.com/adobe/react-spectrum/issues/2296#issue-990465496)

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).

## 📝 Test Instructions:
parseColor() will now support alpha hex values without breaking backwards compatibility

parseColor('#abcd')
parseColor('#aabbccdd')

### CODE REVIEW

**Comment by spsDrop** [comment]
close to fix CLA signature

File: packages/@react-stately/color/src/Color.ts [hunk 1]
```diff
@@ -81,42 +81,33 @@ abstract class Color implements IColor {
...
-        return new RGBColor(r, g, b, 1);
-      } else if (m[2]) {
-        let r = parseInt(m[2][0] + m[2][1], 16);
-        let g = parseInt(m[2][2] + m[2][3], 16);
-        let b = parseInt(m[2][4] + m[2][5], 16);
-        return new RGBColor(r, g, b, 1);
+  static parse(value: string) {
+    let colors = [];
+    // matching #rgb, #rgba, #rrggbb, #rrggbbaa 
+    if (/^#[0-9a-f]+$/i.test(value) && [4, 5, 7, 9].includes(value.length)) {
```
Comment (snowystinger) [inline packages/@react-stately/color/src/Color.ts:92]:
```suggestion
    if (/^#[\da-f]+$/i.test(value) && [4, 5, 7, 9].includes(value.length)) {
```

File: packages/@react-stately/color/src/Color.ts [hunk 2]
```diff
@@ -81,42 +81,33 @@ abstract class Color implements IColor {
...
+        colors.push(parseInt(values.splice(0, 2).join(''), 16));
       }
+      colors[3] = colors[3] !== undefined ? colors[3] / 255 : undefined;
     }
 
-    if ((m = value.match(RGB_REGEX))) {
-      const [r, g, b, a] = (m[1] ?? m[2]).split(',').map(n => Number(n.trim()));
-      return new RGBColor(clamp(r, 0, 255), clamp(g, 0, 255), clamp(b, 0, 255), clamp(a ?? 1, 0, 1));
+    // matching rgb(rrr, ggg, bbb), rgba(rrr, ggg, bbb, 0.a)
+    const match = value.match(/^rgb[a]?\((.*)\)$/);
```
Comment (snowystinger) [inline packages/@react-stately/color/src/Color.ts:101]:
```suggestion
    const match = value.match(/^rgba?\((.*)\)$/);
```

**Review by snowystinger — COMMENTED** [review]
Couple of real small improvements
Thanks for making the PR, I was looking at HSL and HSB, both support parsing the alpha channel, so this brings rgba into line with that.

File: packages/@react-stately/color/src/Color.ts (hunk 2 above)
Comment (spsDrop) [reply packages/@react-stately/color/src/Color.ts]: Implemented the changes

**Review by snowystinger — APPROVED** [review]

**Review by reidbarber — APPROVED** [review]
LGTM

**Comment by snowystinger** [comment]
Noticed during testing today, this does not enable RGBA in ColorField itself due to this line https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/color/src/useColorFieldState.ts#L155
is that something you needed? the PR only says for the color parser, so I'm assuming you aren't using the Field?

**Pull request merged**